src/main/java/net/neoforged/neoforge/event/entity/living/LivingFreezeEvent.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.event.entity.living;
+
+import net.minecraft.world.entity.LivingEntity;
+import net.neoforged.bus.api.ICancellableEvent;
+
+public class LivingFreezeEvent extends LivingEvent {
+    private boolean isFreezing;
+    private int ticksRequiredToFreeze;
+    private float slowAmount;
+    private float damageAmount;
+    private int damageTickRate; // In ticks
```

Comment:
It occured to me that for breathing/drowning, there are two events fired: `LivingBreatheEvent` and `LivingDrownEvent`. Should we have the same distinction for this case?

Reply:
Split in https://github.com/neoforged/NeoForge/pull/2490/commits/70e22439bfdce6d76d4401aa5ca08ce7072789c3